pcdsdevices/pim.py

```diff
@@ -30,39 +28,197 @@ class PIMMotor(InOutRecordPositioner):
     _icon = 'fa.camera-retro'
 
     def stage(self):
-        """
-        Save the original position to be restored on `unstage`.
-        """
+        """Save the original position to be restored on `unstage`."""
         self._original_vals[self.state] = self.state.value
         return super().stage()
 
 
-class PIM(PIMMotor):
+class PIM(Device):
     """
-    Profile intensity monitor, fully motorized and with a detector.
+    Profile intensity monitor with y-motion motor, zoom motor, and a detector.
 
     Parameters
     ----------
     prefix : str
-        The EPICS base of the motor
+        The EPICS base of the PIM
 
     name : str
         A name to refer to the device
 
     prefix_det : str, optional
-        The EPICS base PV of the detector. If None, it will be inferred from
-        the motor prefix
+        The EPICS base PV of the detector. If None, it will be attemptted to be
+        inferred from `prefix`
+
+    prefix_zoom : str, optional
+        The EPICS base PV of the zoom motor. If None, it will be attempted to
+        be inferred from `prefix`
     """
-    detector = FCpt(PCDSAreaDetector, "{self._prefix_det}", kind='normal')
-    tab_whitelist = ["detector"]
-
-    def __init__(self, prefix, *, name, prefix_det=None, **kwargs):
-        # Infer the detector PV from the motor PV
-        if not prefix_det:
-            self._section = prefix.split(":")[0]
-            self._imager = prefix.split(":")[1]
-            self._prefix_det = "{0}:{1}:CVV:01".format(
-                self._section, self._imager)
-        else:
+
+    _prefix_start = ''
+
+    state = Cpt(PIM_Y, '', kind='normal')
+    zoom_motor = FCpt(IMS, '{self._prefix_zoom}', kind='normal')
+    detector = FCpt(PCDSAreaDetector, '{self._prefix_det}', kind='normal')
+
+    tab_whitelist = ['y_motor', 'zoom_motor', 'detector']
+
+    def infer_prefix(self, prefix):
+        """Pulls out the first two segments of the prefix PV, if not already
+           done"""
+        if not self._prefix_start:
+            self._prefix_start = '{0}:{1}:'.format(prefix.split(':')[0],
+                                                   prefix.split(':')[1])
+
+    @property
+    def prefix_start(self):
+        """Returns the first two segments of the prefix PV."""
+        return str(self._prefix_start)
+
+    @property
+    def removed(self):
+        """Returns ``True`` if the yag and diode are removed from the beam."""
+        return self.state.removed
+
+    @property
+    def inserted(self):
+        """Returns ``True`` if yag or diode are inserted."""
+        return self.state.inserted
+
+    def insert(self, moved_cb=None, timeout=None, wait=False):
+        """Moves the diode into the beam."""
```

Comment:
Is this true? Does insert pick the diode over the yag?

Reply:
If so I think the yag gets more use

Reply:
No, it goes to YAG. Docstring was wrong. Now fixed.